lib/rage/router/backend.rb

```diff
@@ -63,6 +68,7 @@ def on(method, path, handler, constraints: {}, defaults: nil)
       raise "Invalid route handler format, expected to match the 'controller#action' pattern" unless handler =~ STRING_HANDLER_REGEXP
 
       controller, action = to_controller_class($1), $2
+      return if controller.nil? && Rage.code_loader.reloading_enabled?
       run_action_method_name = controller.__register_action(action.to_sym)
```

Comment:
It looks like we also need another `return` after this line to prevent the router from creating an incorrect handler.